frame/balances/ring/src/lib.rs

```diff
@@ -530,122 +477,137 @@ decl_module! {
 		) {
 			let transactor = ensure_signed(origin)?;
 			let dest = T::Lookup::lookup(dest)?;
-			<Self as Currency<_>>::transfer(&transactor, &dest, value, ExistenceRequirement::KeepAlive)?;
+			<Self as Currency<_>>::transfer(&transactor, &dest, value, KeepAlive)?;
 		}
-
 	}
 }
 
 impl<T: Trait<I>, I: Instance> Module<T, I> {
 	// PRIVATE MUTABLES
 
-	/// Set the reserved balance of an account to some new value. Will enforce `ExistentialDeposit`
-	/// law, annulling the account as needed.
-	///
-	/// Doesn't do any preparatory work for creating a new account, so should only be used when it
-	/// is known that the account already exists.
-	///
-	/// NOTE: LOW-LEVEL: This will not attempt to maintain total issuance. It is expected that
-	/// the caller will do this.
-	fn set_reserved_balance(who: &T::AccountId, balance: T::Balance) -> UpdateBalanceOutcome {
-		if balance < T::ExistentialDeposit::get() {
-			<ReservedBalance<T, I>>::insert(who, balance);
-			Self::on_reserved_too_low(who);
-			UpdateBalanceOutcome::AccountKilled
-		} else {
-			<ReservedBalance<T, I>>::insert(who, balance);
-			UpdateBalanceOutcome::Updated
-		}
+	/// Get the free balance of an account.
+	pub fn free_balance(who: impl Borrow<T::AccountId>) -> T::Balance {
+		Self::account(who.borrow()).free
 	}
 
-	/// Set the free balance of an account to some new value. Will enforce `ExistentialDeposit`
-	/// law, annulling the account as needed.
-	///
-	/// Doesn't do any preparatory work for creating a new account, so should only be used when it
-	/// is known that the account already exists.
-	///
-	/// NOTE: LOW-LEVEL: This will not attempt to maintain total issuance. It is expected that
-	/// the caller will do this.
-	fn set_free_balance(who: &T::AccountId, balance: T::Balance) -> UpdateBalanceOutcome {
-		// Commented out for now - but consider it instructive.
-		// assert!(!Self::total_balance(who).is_zero());
-		// assert!(Self::free_balance(who) > T::ExistentialDeposit::get());
-		if balance < T::ExistentialDeposit::get() {
-			<FreeBalance<T, I>>::insert(who, balance);
-			Self::on_free_too_low(who);
-			UpdateBalanceOutcome::AccountKilled
-		} else {
-			<FreeBalance<T, I>>::insert(who, balance);
-			UpdateBalanceOutcome::Updated
+	/// Get the frozen balance of an account.
+	fn frozen_balance(who: impl Borrow<T::AccountId>) -> FrozenBalance<T::Balance> {
```

Comment:
suggest to renaming to ```extended_locked_balance``` to avoid terminology confuse, because there is another ```frozen(``` method in FrozenBalance.

Reply:
`frozen_balance` looks good. `frozen` need a reason param, change the `frozen` to `frozen_for`?